src/model/OutputData.ts

```diff
@@ -92,58 +108,84 @@ export class OutputData implements OutputDataLike {
 		return amounts.map((a) => this.createSingleP2PKData(p2pk, a, keyset.id));
 	}
 
-	static createSingleP2PKData(
-		p2pk: {
-			pubkey: string | string[];
-			locktime?: number;
-			refundKeys?: string[];
-			requiredSignatures?: number;
-			requiredRefundSignatures?: number;
-		},
-		amount: number,
-		keysetId: string,
-	) {
-		// Standardize pubkey (backwards compat), clamp n_sigs between 1 and total pubkeys
-		// clamp n_sigs_refund between 1 and total refundKeys, and create secret
-		const pubkeys: string[] = Array.isArray(p2pk.pubkey) ? p2pk.pubkey : [p2pk.pubkey];
-		const n_sigs: number = Math.max(1, Math.min(p2pk.requiredSignatures || 1, pubkeys.length));
-		const n_sigs_refund: number = Math.max(
+	static createSingleP2PKData(p2pk: P2PKOptions, amount: number, keysetId: string) {
+		// normalise keys and clamp required signature counts to available keys
+		const lockKeys: string[] = Array.isArray(p2pk.pubkey) ? p2pk.pubkey : [p2pk.pubkey];
+		const refundKeys: string[] = p2pk.refundKeys ?? [];
+		const reqLock = Math.max(1, Math.min(p2pk.requiredSignatures ?? 1, lockKeys.length));
+		const reqRefund = Math.max(
 			1,
-			Math.min(p2pk.requiredRefundSignatures || 1, p2pk.refundKeys ? p2pk.refundKeys.length : 1),
+			Math.min(p2pk.requiredRefundSignatures ?? 1, refundKeys.length || 1),
 		);
+
+		// Init vars
+		const data = lockKeys[0];
+		const pubkeys = lockKeys.slice(1);
+		const refund = refundKeys;
+
+		// build P2PK Tags (NUT-11)
+		const tags: string[][] = [];
+
+		if (p2pk.locktime !== undefined) {
```

Comment:
```suggestion
		if (typeof p2pk.locktime === 'number') {
```

This guards from null passing this check (useful for JS consumers or consumers that pass in JSON parsed data without validation)

Reply:
Thank you! I think more validation is not strictly necessary, but when we are at it anyways, why not? What about adding a check for valid Date?

Number.isSafeInteger(ls) && ts >= 0 && !isNaN(new Date(ts * 1000))

I think we can get rid of the typeof check because `isSafeInteger` will take care of that too